### PR TITLE
Reformat "Important Files" section in `compiler.rst`

### PR DESCRIPTION
**Pull request by ezio-melotti**
This is a follow-up of #984 that reformats the "Important Files" section in `compiler.rst`:

> I also found another issue with this section: https://devguide.python.org/internals/compiler/#important-files
> In addition to the fact that the listed files don't use any markup, the markup used for most of the files seem incorrect (it's a blockquote due to the indentation). A nested list with full paths for all the files and :cpy-file: should be a better alternative.

In the PR I added links to the files using `:cpy-file:` and reformatted the list.

[Before](https://devguide.python.org/internals/compiler/#important-files):
image
After:
image

### CODE REVIEW

File: internals/compiler.rst [hunk 1]
```diff
@@ -524,140 +524,112 @@ statement in ``_PyEval_EvalFrameDefault()``.
...
-        Emits bytecode based on the AST.
+   * :cpy-file:`Python/peephole.c`: Optimizes the bytecode.
 
-    symtable.c
-        Generates a symbol table from AST.
+   * :cpy-file:`Python/pyarena.c`: Implementation of the arena memory manager.
 
-    peephole.c
-        Optimizes the bytecode.
+   * :cpy-file:`Python/wordcode_helpers.h`: Helpers for generating bytecode.
```
Comment (hugovk) [inline internals/compiler.rst:586]: https://github.com/python/cpython/blob/main/Python/wordcode_helpers.h

[This file](https://github.com/python/cpython/blob/18b5dd68c6b616257ae243c0b6bb965ffc885a23/Python%2Fwordcode_helpers.h) was deleted ([on Feb 25](https://github.com/python/cpython/commit/0f41aac109c45aa468c432f798947c54d4178b3d)) - [Commit history](https://github.com/python/cpython/commits/main/Python/wordcode_helpers.h).

File: internals/compiler.rst [hunk 2]
```diff
@@ -524,140 +524,112 @@ statement in ``_PyEval_EvalFrameDefault()``.
...
-        (for string annotations).
+   * :cpy-file:`Python/compile.c`: Emits bytecode based on the AST.
 
-    ceval.c
-        Executes byte code (aka, eval loop).
+   * :cpy-file:`Python/symtable.c`: Generates a symbol table from AST.
 
-    compile.c
-        Emits bytecode based on the AST.
+   * :cpy-file:`Python/peephole.c`: Optimizes the bytecode.
```
Comment (hugovk) [inline internals/compiler.rst:582]: https://github.com/python/cpython/blob/main/Python/peephole.c

[This file](https://github.com/python/cpython/blob/ba18c0b13ba3c08077ea3db6658328523823a33f/Python%2Fpeephole.c) was deleted ([on Jul 30, 2020](https://github.com/python/cpython/commit/6e8128f02e1d36e38e5866f9dc36e051caa47bc9)) - [Commit history](https://github.com/python/cpython/commits/main/Python/peephole.c).

File: internals/compiler.rst [hunk 3]
```diff
@@ -524,140 +524,112 @@ statement in ``_PyEval_EvalFrameDefault()``.
...
 
-    pyarena.c
-        Implementation of the arena memory manager.
+   * :cpy-file:`Python/opcode_targets.h`: One of the files that must be
+     modified if :cpy-file:`Lib/opcode.py` is.
 
-    wordcode_helpers.h
-        Helpers for generating bytecode.
+* :cpy-file:`Include/`
+   * :cpy-file:`Include/code.h`: Header file for :cpy-file:`Objects/codeobject.c`;
```
Comment (hugovk) [inline internals/compiler.rst:592]: https://github.com/python/cpython/blob/main/Include/code.h

[This file](https://github.com/python/cpython/blob/5c4d1f6e0e192653560ae2941a6677fbf4fbd1f2/Include%2Fcode.h) was deleted ([on Apr 7](https://github.com/python/cpython/commit/85addfb9c6496eb3d26082348cf5aca848c877ef)) - [Commit history](https://github.com/python/cpython/commits/main/Include/code.h).

**Review by hugovk — COMMENTED** [review]
Couple more deleted files, the rest look okay.

Otherwise looks good, thanks!

**Comment by ezio-melotti** [comment]
The deleted files are tracked in #1003.

**Review by hugovk — APPROVED** [review]

**Comment by CAM-Gerlach** [comment]
If we're reformatting the whole section like this, maybe worth fixing the lexicographic order too while we're at it?

**Comment by ezio-melotti** [comment]
I think they are roughly sorted from most to least important.

**Review by CAM-Gerlach — APPROVED** [review]
LGTM then, thanks @ezio-melotti

**Pull request merged**